content/things/observables-on-the-platform.md

```diff
@@ -0,0 +1,375 @@
+
+**TL;DR:** Streams and observables are very similar. Their biggest difference seems to be that streams can deliver their data only once and that streams do so asynchronously. Writing a [demo app][dof tool] with my [stream-based observables library][ows] instead of RxJS makes me think that these differences don’t really matter when it comes to writing web apps. Go try it for yourself!
+
+## Observables? Reactive Programming?
+
+The Reactive Programming (RP) paradigm is not new and has enjoyed popularity in many different genres of programming. For example Android folks, especially since the introduction of Kotlin, have seen a rise in popularity around RP. I have also noticed some game engines experimenting with RP. On the web there have been a number of attempts to enable reactive programming. [RxJS] and [CycleJS] are front of mind to me. More recently, [Svelte 3][svelte] is trying to establish [a slightly different take on reactive programming][svelte 3 reactivity].
+
+Reactive programming in general as well as these specific implementations are well covered on the web, so I won’t write yet another introduction to RP. But I want to make sure we are on the same page. According to [Wikipedia][wiki rp], “reactive programming is a _declarative_ programming paradigm concerned with _data streams_ and the _propagation of change_.” (Emphasis mine.) An “observable” is a type on which you can register “observers”, as defined in the [observer pattern] in [the GoF book]. For the time being, you can think of an observer as a callback in JavaScript land:
+
+```js
+myButton.addEventListener("click", myClickHandler);
+```
+
+In this example, `myClickHandler` would be the observer, while `myButton` is an observable. Kinda. The differences (as I perceive them), are subtle but important:
+
+- An observable is a single stream of data. `myButton` here has multiple “streams”, one for each event type. So one for clicks, one for moving the mouse, one for key presses etc.
+- A stream implies an input, an output and an order. Event listener don’t have an output as the return value of an event listener is discarded. And if the event handler is asynchronous, the next event of the same type can get processed despite the previous event one not having finished processing.
+
+> **Note:** As I said before, there are multiple libraries that employ the RP pattern. For the remainder of this blog post, I am going to focus on [RxJS] to keep the blog post clear and because their documentation was a major source of information for me.
+
+RxJS implements an `Observable` type and also ships many utility functions, for example to turn that button into a proper observable:
+
+```js
+import { fromEvent } from "rxjs";
+
+const observable = fromEvent(myButton, "click");
+observable.subscribe(ev => {
```

Comment:
Maybe rename to `event` throughout the article? Currently it's sometimes `ev`, sometimes `v`.